### PR TITLE
feat: use split store in the view client

### PR DESCRIPTION
**Pull request by wacban**
- use split store in the view client - when configured
- removed the get_store(Temperature) method and replaced usages with safer get_cold_store and get_hot_store
- added SplitStorageConfig and used it to configure a few things

### CODE REVIEW

**Comment by wacban** [comment]
https://nayduck.near.org/#/run/2887

**Comment by wacban** [comment]
https://nayduck.near.org/#/run/2888

**Comment by wacban** [comment]
Some tests were failing in nayduck so I hid the split storage in view client behind a config that by default is false. Also configured a few other split storage things that will be handy to tune tests. 

https://nayduck.near.org/#/run/2891

File: core/store/src/lib.rs [hunk 1]
```diff
@@ -182,12 +182,12 @@ impl NodeStorage {
     ///
     /// This method panics if trying to access cold store but it wasn't configured.
     /// Please consider using the get_hot_store and get_cold_store methods to avoid panics.
-    pub fn get_store(&self, temp: Temperature) -> Store {
-        match temp {
-            Temperature::Hot => self.get_hot_store(),
-            Temperature::Cold => self.get_cold_store().unwrap(),
-        }
-    }
+    // pub fn get_store(&self, temp: Temperature) -> Store {
```
Comment (nikurt) [inline core/store/src/lib.rs:185]: Remove?

**Review by nikurt — APPROVED** [review]

**Pull request merged**